### PR TITLE
Store full remote worker results

### PR DESCRIPTION
**Pull request by msmner**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/734

Linked pull requests from another repos (if any):

**Summary of the changes made**:
1. Add full worker execution results to submissions for processing

### CODE REVIEW

File: Data/OJS.Data.Models/Submissions/SubmissionForProcessing.cs [hunk 1]
```diff
@@ -9,5 +9,13 @@ public class SubmissionForProcessing : AuditInfoEntity<int>
         public bool Processing { get; set; }
 
         public bool Processed { get; set; }
+
+        public string? SerializedExecutionDetails { get; set; }
```
Comment (shsimeonova) [inline Data/OJS.Data.Models/Submissions/SubmissionForProcessing.cs:13]: Remove these changes

File: Data/OJS.Data/Migrations/20230829134213_AddedRemoteWorkerResults.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+﻿#nullable disable
+
+namespace OJS.Data.Migrations
```
Comment (shsimeonova) [inline Data/OJS.Data/Migrations/20230829134213_AddedRemoteWorkerResults.cs:3]: Remove these changes

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs [hunk 3]
```diff
@@ -467,6 +482,11 @@ public async Task ProcessExecutionResult(SubmissionExecutionResult submissionExe
         }
         else
         {
+            if (submissionForProcessing != null)
+            {
+                submissionForProcessing.SerializedException = JsonConvert.SerializeObject(exception);
```
Comment (gogo4ds) [inline Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:487]: Can the exception be `null`?

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs [hunk 4]
```diff
@@ -429,8 +429,16 @@ public async Task Submit(SubmitSubmissionServiceModel model)
         scope.Complete();
         scope.Dispose();
 
+        var submissionToBePublished = this.BuildSubmissionForProcessing(newSubmission, problem, submissionType);
+        var submissionForProcessing = await this.submissionsForProcessingData.GetBySubmission(newSubmission.Id);
+        if (submissionForProcessing != null)
+        {
+            submissionForProcessing.SerializedExecutionDetails = JsonConvert.SerializeObject(submissionToBePublished);
```
Comment (gogo4ds) [inline Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:436]: We have `.ToJson()` extension method

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs [hunk 5]
```diff
@@ -456,9 +464,16 @@ public async Task ProcessExecutionResult(SubmissionExecutionResult submissionExe
         submission.Processed = true;
         submission.ProcessingComment = null;
 
+        var submissionForProcessing = await this.submissionsForProcessingData.GetBySubmission(submission.Id);
```
Comment (gogo4ds) [inline Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:467]: Here also, why get the entity again? I thinkg the entity is retrieved once in `MarkProcessed` method below. Just save the result in the same transaction that updates it to Processed.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs (hunk 4 above)
Comment (msmner) [reply Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:436]: Used.

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs (hunk 5 above)
Comment (msmner) [reply Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:467]: Done.

File: Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs [hunk 6]
```diff
@@ -429,8 +429,16 @@ public async Task Submit(SubmitSubmissionServiceModel model)
         scope.Complete();
         scope.Dispose();
 
+        var submissionToBePublished = this.BuildSubmissionForProcessing(newSubmission, problem, submissionType);
+        var submissionForProcessing = await this.submissionsForProcessingData.GetBySubmission(newSubmission.Id);
```
[the remark this answers is not in the document]
Comment (msmner) [reply Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs:433]: Done.

**Comment by msmner** [comment]
Closed this PR due to issues. New updated PR is https://github.com/SoftUni-Internal/OpenJudgeSystem/pull/834.

**Pull request closed without merging**